src/dpdkstat.c

```diff
@@ -0,0 +1,775 @@
+               child->values[0].value.string);
+      DEBUG("dpdkstat:Memory Channels %s \n", g_configuration->memory_channels);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("SocketMemory", child->key) == 0) {
+      ssnprintf(g_configuration->socket_memory, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: socket mem %s \n", g_configuration->socket_memory);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("ProcessType", child->key) == 0) {
+      ssnprintf(g_configuration->process_type, DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: proc type %s \n", g_configuration->process_type);
+      g_configuration->eal_argc+=1;
+    } else if (strcasecmp("FilePrefix", child->key) == 0) {
+      ssnprintf(g_configuration->file_prefix, DATA_MAX_NAME_LEN, "/var/run/.%s_config",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: file prefix %s \n", g_configuration->file_prefix);
+      if (strcasecmp(g_configuration->file_prefix, "/var/run/.rte_config") != 0) {
+        g_configuration->eal_argc+=1;
+      }
+    } else if (strcasecmp("EnabledPortMask", child->key) == 0) {
+      g_configuration->enabled_port_mask = (uint32_t)child->values[0].value.number;
+      DEBUG("dpdkstat: Enabled Port Mask %u\n", g_configuration->enabled_port_mask);
+    } else if (strcasecmp("PortName", child->key) == 0) {
+      ssnprintf(g_configuration->port_name[port_counter], DATA_MAX_NAME_LEN, "%s",
+               child->values[0].value.string);
+      DEBUG("dpdkstat: Port %d Name: %s \n", port_counter,
+               g_configuration->port_name[port_counter]);
+      port_counter++;
+    } else {
+      WARNING ("dpdkstat: The config option \"%s\" is unknown.",
+               child->key);
+    }
+  } /* End for (int i = 0; i < ci->children_num; i++)*/
+  g_configured = 1; /* Bypass configuration in dpdk_shm_init(). */
+
+  return 0;
+}
+
+/* Initialize SHared Memory (SHM) for config and helper process */
+static int dpdk_shm_init(size_t size)
+{
+  /*
+   * Check if SHM is already configured: when config items are provided, the
+   * config function initializes SHM. If there is no config, then init() will
+   * just return.
+   */
+  if(g_configuration)
+    return 0;
+
+  /* Create and open a new object, or open an existing object. */
+  int fd = shm_open(DPDK_SHM_NAME, O_CREAT | O_TRUNC | O_RDWR, 0666);
+  if (fd < 0) {
+    WARNING("dpdkstat:Failed to open %s as SHM:%s\n", DPDK_SHM_NAME,
+            strerror(errno));
+    goto fail;
+  }
+  /* Set the size of the shared memory object. */
+  int ret = ftruncate(fd, size);
+  if (ret != 0) {
+    WARNING("dpdkstat:Failed to resize SHM:%s\n", strerror(errno));
+    goto fail_close;
+  }
+  /* Map the shared memory object into this process' virtual address space. */
+  g_configuration = mmap(0, size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
```

Comment:
This function always unconditionally assigns the pointer to `g_configuration` but gets the size via an argument. Can we remove the `size` argument and use `sizeof (*g_configuration)` instead?